File: VB/E2032/Converters.vb
Imports System
Imports System.Globalization
Imports System.Linq
Imports System.Windows.Controls
Imports System.Windows.Data
Imports System.Windows.Media
Imports System.Windows.Media.Imaging

Namespace E2032
	Public Class RemoveParenthesesConverter
		Implements IValueConverter

		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
			Dim str = TryCast(value, String)
			If str Is Nothing Then
				Return Nothing
			End If
			Return str.Replace("(", String.Empty).Replace(")", String.Empty)
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class

	Public Class PrintRowInfoToImageSourceConverter
		Implements IValueConverter

		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
			Dim category As Category = Nothing
			If System.Enum.TryParse(value.ToString(), category) Then
				Select Case category
					Case E2032.Category.Deferred
						Return "Images/Deffered_32x32.png"
					Case E2032.Category.Normal
						Return "Images/Normal_32x32.png"
					Case E2032.Category.Urgent
						Return "Images/Urgent_32x32.png"
				End Select
			End If
			Return Nothing
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class
End Namespace

File: CS/E2032/Converters.cs
using System;
using System.Globalization;
using System.Linq;
using System.Windows.Data;

namespace E2032 {
    public class RemoveParenthesesConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            var str = value as string;
            if (str == null)
                return null;
            return str.Replace("(", string.Empty).Replace(")", string.Empty);
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }

    public class PrintRowInfoToImageSourceConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            Category category;
            if (Enum.TryParse(value.ToString(), out category)) {
                switch (category) {
                    case Category.Deferred:
                        return "Deffered_32x32.png";
                    case Category.Normal:
                        return "Normal_32x32.png";
                    case Category.Urgent:
                        return "Urgent_32x32.png";
                }
            }
            return null;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}
